### PR TITLE
Update autorest.csharp to 3.0.0-beta.20220113.5

### PR DESCRIPTION
**Pull request by AlexanderSher**
PR consists of several commits, one project per commit

### CODE REVIEW

File: sdk/websites/Azure.ResourceManager.AppService/src/Generated/WebSiteCollection.cs [hunk 1]
```diff
@@ -276,7 +276,7 @@ public async virtual Task<Response<bool>> ExistsAsync(string name, CancellationT
                 throw new ArgumentNullException(nameof(name));
             }
 
-            using var scope = _clientDiagnostics.CreateScope("WebSiteCollection.ExistsAsync");
+            using var scope = _clientDiagnostics.CreateScope("WebSiteCollection.Exists");
```
Comment (ArthurMa1978) [inline sdk/websites/Azure.ResourceManager.AppService/src/Generated/WebSiteCollection.cs]: Why change this from Async to sync?

File: sdk/websites/Azure.ResourceManager.AppService/src/Generated/WebSiteCollection.cs [hunk 2]
```diff
@@ -224,14 +224,14 @@ public async virtual Task<Response<WebSite>> GetIfExistsAsync(string name, Cance
                 throw new ArgumentNullException(nameof(name));
             }
 
-            using var scope = _clientDiagnostics.CreateScope("WebSiteCollection.GetIfExistsAsync");
+            using var scope = _clientDiagnostics.CreateScope("WebSiteCollection.GetIfExists");
```
Comment (ArthurMa1978) [inline sdk/websites/Azure.ResourceManager.AppService/src/Generated/WebSiteCollection.cs]: Why change this from Async to sync?

File: sdk/websites/Azure.ResourceManager.AppService/src/Generated/WebSiteCollection.cs (hunk 1 above)
Comment (Yao725) [reply sdk/websites/Azure.ResourceManager.AppService/src/Generated/WebSiteCollection.cs:279]: It's a minor fix to keep the scope name consistent, in other methods like [`CreateOrUpdateAsync`](https://github.com/Azure/azure-sdk-for-net/blob/d9227a57be17cde5697c6d48dca986d23b347c70/sdk/compute/Azure.ResourceManager.Compute/src/Generated/AvailabilitySetCollection.cs#L106) we always use the name without 'Async" suffix.

File: sdk/websites/Azure.ResourceManager.AppService/src/Generated/WebSiteCollection.cs (hunk 2 above)
Comment (Yao725) [reply sdk/websites/Azure.ResourceManager.AppService/src/Generated/WebSiteCollection.cs:227]: same reason as above

File: sdk/storage/Azure.ResourceManager.Storage/tests/Tests/StorageAccountTests.cs [hunk 3]
```diff
@@ -1187,11 +1187,15 @@ public async Task CreateStorageAccountWithEnableNfsV3()
 
         [Test]
         [RecordedTest]
+        [Ignore("Looks like a regression, temporarily removed")]
```
Comment (Yao725) [inline sdk/storage/Azure.ResourceManager.Storage/tests/Tests/StorageAccountTests.cs]: This regression should be a corner case after introducing this [autorest change](https://github.com/Azure/autorest.csharp/pull/1838), so temporarily remove this test, will do more investigation.

File: sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/Generated/ApplicationDataClient.cs [hunk 4]
```diff
@@ -119,6 +113,9 @@ public ApplicationDataClient(Uri endpoint, TokenCredential credential, FarmBeats
         public virtual async Task<Response> GetApplicationDataAsync(string farmerId, string applicationDataId, RequestContext context = null)
 #pragma warning restore AZC0002
         {
+            Argument.AssertNotNull(farmerId, nameof(farmerId));
```
Comment (annelo-msft) [inline sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/Generated/ApplicationDataClient.cs:116]: Do we know that null isn't a valid value for these?

**Review by ArthurMa1978 — APPROVED** [review]

File: sdk/appconfiguration/Azure.ResourceManager.AppConfiguration/api/Azure.ResourceManager.AppConfiguration.netstandard2.0.cs [hunk 5]
```diff
@@ -15,8 +15,8 @@ public partial class ConfigurationStore : Azure.ResourceManager.Core.ArmResource
...
         public static Azure.Core.ResourceIdentifier CreateResourceIdentifier(string subscriptionId, string resourceGroupName, string configStoreName) { throw null; }
-        public virtual Azure.ResourceManager.AppConfiguration.Models.ConfigurationStoreDeleteOperation Delete(bool waitForCompletion = true, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.ResourceManager.AppConfiguration.Models.ConfigurationStoreDeleteOperation> DeleteAsync(bool waitForCompletion = true, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.ResourceManager.AppConfiguration.Models.ConfigurationStoreDeleteOperation Delete(bool waitForCompletion, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (annelo-msft) [inline sdk/appconfiguration/Azure.ResourceManager.AppConfiguration/api/Azure.ResourceManager.AppConfiguration.netstandard2.0.cs:18]: Curious why this is coming in as part of this change?

File: sdk/webpubsub/Azure.Messaging.WebPubSub/src/Generated/WebPubSubServiceClient.cs [hunk 6]
```diff
@@ -216,6 +216,8 @@ public virtual Response CloseAllConnections(IEnumerable<string> excluded = null,
         public virtual async Task<Response> SendToAllAsync(RequestContent content, ContentType contentType, IEnumerable<string> excluded = null, RequestContext context = null)
 #pragma warning restore AZC0002
         {
+            Argument.AssertNotNull(content, nameof(content));
```
Comment (annelo-msft) [inline sdk/webpubsub/Azure.Messaging.WebPubSub/src/Generated/WebPubSubServiceClient.cs]: It is a breaking change to introduce an exception where there wasn't one before, and this library has GA'ed

Comment (m-nash) [reply sdk/webpubsub/Azure.Messaging.WebPubSub/src/Generated/WebPubSubServiceClient.cs]: @AlexanderSher ^^

File: sdk/storage/Azure.ResourceManager.Storage/tests/Tests/StorageAccountTests.cs (hunk 3 above)
Comment (blueww) [reply sdk/storage/Azure.ResourceManager.Storage/tests/Tests/StorageAccountTests.cs]: This should be caused by the swagger/server are not aligned on the DeletedAccounts API. 
Swagger/Server both owned by SRP team.
SRP team has confirmed in mail: 
> We have a work item dealing with swagger correctness and the highlighted issue will be addressed in the next api version release.
